src/hotspot/share/opto/inlinetypenode.cpp

```diff
@@ -742,7 +847,8 @@ bool InlineTypeNode::is_default(PhaseGVN* gvn) const {
     } else if (value->is_InlineType()) {
       value = value->as_InlineType()->get_oop();
     }
-    if (!gvn->type(value)->is_zero_type()) {
+    if (!gvn->type(value)->is_zero_type() &&
+        !VectorNode::is_all_zeros_vector(value)) {
```

Comment:
Better to merge `VectorNode::is_all_zeros_vector(value)` with `is_zero_type()`. But it seems we have to add the constants for vector type first. Not a block.